src/core/Global.ttslua

```diff
@@ -838,15 +859,23 @@ function onClick_toggleOption(_, id)
   applyOptionPanelChange(id, state)
 end
 
+-- called by the language selection dropdown
+function languageSelected(_, selectedIndex, id)
+  optionPanel[id] = tonumber(selectedIndex)
+end
+
 -- sets the option panel to the correct state (corresponding to 'optionPanel')
 function updateOptionPanelState()
   for id, enabled in pairs(optionPanel) do
```

Comment:
enabled has become disconnected as it describes a boolean but can actually be pretty much any type.  Consider a name change for clarity.  value or optionValue?